src/config/typeorm.config.ts

```diff
@@ -32,31 +32,74 @@ export class TypeOrmConfigService implements TypeOrmOptionsFactory {
       ? false
       : rawLogging.split(',').map(level => level.trim()) as LoggerOptions;
 
-    return {
-      type: 'postgres',
-      applicationName: this.configService.get<string>('app.name'),
-      host: this.configService.get<string>('database.host'),
-      port: this.configService.get<number>('database.port'),
-      username: this.configService.get<string>('database.user'),
-      password: this.configService.get<string>('database.pwd'),
-      database: this.configService.get<string>('database.name'),
+    const baseConfig = {
+      type: 'postgres' as const,
       entities: [__dirname + '/../**/*.entity.{js,ts}'],
       synchronize: false,
-      ssl: this.tlsOptions,
-
-      // Database specific (Postgres) settings.
-      extra: {
-        max: this.configService.get<number>('app.maxConnectionPool'),                                 // Max connections in pool
-        idleTimeoutMillis: this.configService.get<number>('app.idleTimeout'),                         // Close idle connections
-        connectionTimeoutMillis: this.configService.get<number>('app.connectionTimeout'),             // Maximum time (ms) to wait for a new connection before timing out.
-        statement_timeout: this.configService.get<number>('app.statementTimeout'),                    // Terminates queries that exceed the timeout (in ms).
-        idle_in_transaction_session_timeout: this.configService.get<number>('app.idleInTransactionSessionTimeout'), // Terminates idle transactions after the specified time (in ms).
-        maxUses: this.configService.get<number>('app.maxUsesBeforeRecreatingConnection'), //Recreate connections after 'n' uses
-      },
       logging: sqlLogging,
-      // Logs queries exceeding this limit (does not terminate, 'statement_timeout' terminates them).
       maxQueryExecutionTime: this.configService.get<number>('app.maxQueryExecutionTime'),
```

Comment:
The `extra` options should go here:
```suggestion
      maxQueryExecutionTime: this.configService.get<number>('app.maxQueryExecutionTime'),
      extra: {
        max: this.configService.get<number>('app.maxConnectionPool'),
        idleTimeoutMillis: this.configService.get<number>('app.idleTimeout'),
        connectionTimeoutMillis: this.configService.get<number>('app.connectionTimeout'),
        statement_timeout: this.configService.get<number>('app.statementTimeout'),
        idle_in_transaction_session_timeout: this.configService.get<number>('app.idleInTransactionSessionTimeout'),
        maxUses: this.configService.get<number>('app.maxUsesBeforeRecreatingConnection'),
      },
```